specification/ai/OpenAI.Assistants/runs/models.tsp

```diff
@@ -0,0 +1,187 @@
+model RunError {
+  @doc("The status for the error.")
+  code: string;
+
+  @doc("The human-readable text associated with the error.")
+  message: string;
+}
+
+@doc("Possible values for the status of an assistant thread run.")
+@added(ServiceApiVersions.v2023_11_06_beta)
+enum RunStatus {
+  @doc("Represents a run that is queued to start.")
+  queued: "queued",
+
+  @doc("Represents a run that is in progress.")
+  inProgress: "in_progress",
+
+  @doc("Represents a run that needs another operation, such as tool output submission, to continue.")
+  requiresAction: "requires_action",
+
+  @doc("Represents a run that is in the process of cancellation.")
+  cancelling: "cancelling",
+
+  @doc("Represents a run that has been cancelled.")
+  cancelled: "cancelled",
+
+  @doc("Represents a run that failed.")
+  failed: "failed",
+
+  @doc("Represents a run that successfully completed.")
+  completed: "completed",
+
+  @doc("Represents a run that expired before it could otherwise finish.")
+  expired: "expired",
+}
+
+@discriminator("type")
+@doc("An abstract representation of a required action for an assistant thread run to continue.")
+@added(ServiceApiVersions.v2023_11_06_beta)
+model RequiredAction {
+  @doc("The object type.")
+  type: string;
+}
+
+@doc("The details for required tool calls that must be submitted for an assistant thread run to continue.")
+@added(ServiceApiVersions.v2023_11_06_beta)
+model SubmitToolOutputsAction extends RequiredAction {
+  @doc("The object type, which is always 'submit_tool_outputs'.")
+  type: "submit_tool_outputs";
+
+  @projectedName("json", "submit_tool_outputs")
+  @doc("The details describing tools that should be called to submit tool outputs.")
+  submitToolOutputs: SubmitToolOutputsDetails;
+}
+
+@doc("The details describing tools that should be called to submit tool outputs.")
+@added(ServiceApiVersions.v2023_11_06_beta)
+model SubmitToolOutputsDetails {
+  @projectedName("json", "tool_calls")
+  @doc("The list of tool calls that must be resolved for the assistant thread run to continue.")
+  toolCalls: ToolCall[];
+}
+
+@doc("The details used when creating and immediately running a new assistant thread.")
+@added(ServiceApiVersions.v2023_11_06_beta)
+model CreateAndRunThreadOptions {
+  @projectedName("json", "assistant_id")
+  @doc("The ID of the assistant for which the thread should be created.")
+  assistantId: string;
+
+  @doc("The details used to create the new thread.")
+  thread?: AssistantThreadCreationOptions;
+
+  @doc("The overridden model that the assistant should use to run the thread.")
+  @projectedName("csharp", "overrideModelName")
+  `model`?: string;
+
+  @doc("The overridden system instructions the assistant should use to run the thread.")
+  @projectedName("csharp", "overrideInstructions")
+  instructions?: string;
+
+  @doc("The overridden list of enabled tools the assistant should use to run the thread.")
+  @projectedName("csharp", "overrideTools")
+  tools?: ToolDefinition[];
```

Comment:
I see `model`, `instructions` and `tools` as nullable ( `| null` ), ref lines 7733, 7737 and 7740 from the openAPI yaml